sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c

```diff
@@ -358,6 +374,14 @@ void gnrc_lorawan_event_no_rx(gnrc_lorawan_t *mac)
         return;
     }
 
+    if (mac->mlme.adr_ack_cnt  > (CONFIG_LORAMAC_DEFAULT_ADR_ACK_LIMIT +
+         CONFIG_LORAMAC_DEFAULT_ADR_ACK_DELAY)) {
+             if (mac->last_dr) {
+                DEBUG("gnrc_lorawan_mcps: ADRACKReq: Decrement DR\n");
+                mac->last_dr--;
+             }
+    }
```

Comment:
This snippet is still wrong...
as it is, it decreases the DR anytime there's no RX and the `adr_ack_cnt` is greater than `CONFIG_LORAMAC_DEFAULT_ADR_ACK_LIMIT` + `CONFIG_LORAMAC_DEFAULT_ADR_ACK_DELAY`.

This pattern was already proposed in https://github.com/RIOT-OS/RIOT/pull/15995#discussion_r583743697. Is there anything wrong with that?

Reply:
Oops. Thanks for pointing this out 👍🏼